metis-enrichment/metis-enrichment-common/src/main/java/eu/europeana/enrichment/api/internal/AbstractSearchTerm.java

```diff
@@ -1,46 +1,45 @@
 package eu.europeana.enrichment.api.internal;
 
-import eu.europeana.metis.schema.jibx.LanguageCodes;
 import java.util.Objects;
 
+/**
+ * This class is a basic implementation of {@link SearchTerm} that leaves the details of the
+ * candidate types unimplemented.
+ */
 public abstract class AbstractSearchTerm implements SearchTerm {
 
   private final String textValue;
-  private final LanguageCodes language;
+  private final String language;
 
-  public AbstractSearchTerm(String textValue, String language){
+  public AbstractSearchTerm(String textValue, String language) {
     this.textValue = textValue;
-    this.language = LanguageCodes.convert(language);
+    this.language = language;
   }
 
   @Override
-  public boolean equals(Object other){
-    if(other == this){
+  public final boolean equals(Object otherObject) {
```

Comment:
Same here:

"Why final here, were are enforcing equals/hashcode in the hierarchy which I believe we should not"